common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -1864,34 +1871,26 @@ private void finalizePartition() throws IOException {
       metaFile.getChannel().truncate(metaFile.getPos());
     }
 
-    void closeAllFilesAndDeleteIfNeeded(boolean delete) {
+    private void deleteAllFiles() {
       try {
-        if (dataChannel.isOpen()) {
-          dataChannel.close();
-        }
-        if (delete) {
-          dataFile.delete();
+        if (!dataFile.delete()) {
+          logger.warn("Error deleting data file for {} reduceId {}",
+            appAttemptShuffleMergeId, reduceId);
         }
-      } catch (IOException ioe) {
-        logger.warn("Error closing data channel for {} reduceId {}",
+      } catch (Exception exception) {
```

Comment:
hmm... In what scenarios would `dataFile.delete()` throw an exception?

Reply:
It may be throw a SecurityException

Reply:
OK